qutebrowser/browser/webengine/webenginetab.py

```diff
@@ -201,70 +201,112 @@ class WebEngineCaret(browsertab.AbstractCaret):
 
     @pyqtSlot(usertypes.KeyMode)
     def _on_mode_entered(self, mode):
-        pass
+        self._tab.run_js_async(
+            javascript.assemble('caret', 'setInitialCursor'))
```

Comment:
You do `self._tab.run_js_async(javascript.assemble('caret', ...)))` a lot here. Why not add some `_js_call` method or so to simplify those?